### ISSUE TITLE
検証+採用判断: parameterized副官・埋札policyを独立holdoutで再評価し正式artifact候補化する

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#454]
Related: #202, #452
Base branch: `feature/issue-202-non-playing-ai`

## 背景

#452 で、副官35 + 埋札60 = 95 parameter のvisible-only評価関数を pycma CMA-ES で joint optimize した。

固定:
- bidding: `frozen-raise-v1`
- playing: `ppo-separated-v1000`
- 学習対象: 副官 + 埋札のみ
- fitness: Napoleon candidate の mean relative reward
- generation内candidate比較: Common Random Numbers (同一deal seeds)

#452 final holdout 5,000 paired games:
- learned mean relative reward: `1.14595`
- existing RuleBased: `-0.06705`
- paired difference: `+1.21300`
- SE: `0.25256`
- 95% CI: `[0.71799, 1.70801]`
- contract success: `34.12% vs 30.58%`
- contract margin: `-2.0702 vs -2.5266`
- illegal/fallback: `0 / 0`

これは明確な改善だが、方針転換後の最初の本成功なので、runtime wiring / Frozen化の前に完全独立seedでもう一度再現性を確認する。

## 目的

1. #452 best parameter vector を一切再学習せず固定する
2. #452 train / validation / final で未使用の新しい独立seed poolを作る
3. existing RuleBased副官+埋札と、同じseedでpaired比較する
4. 追加5k〜10k gamesで #452 の改善が再現するか確認する
5. 再現した場合、parameterized副官+埋札policyを正式artifact候補としてrepo-managed保存する
6. 次Issueでruntime wiring / Frozen化へ進むべきか判断する

## 最重要: 再学習禁止

このIssueは再評価Issue。

- CMA-ESを再開しない
- parameterを微調整しない
- feature追加しない
- validationを見てparameterを選び直さない

#452で選ばれたbest parameter vectorをそのまま評価する。

これにより追加holdoutを真の独立検証として扱う。

## 評価seed

新規final-verification seed poolは、#452の:
- train generation seeds
- validation seeds
- final 5,000 seeds

のすべてとoverlap 0を保証する。

seed manifest/hashを保存する。

推奨:
```text
10,000 paired games
```

計算時間が大きい場合でも最低5,000 paired gamesは実施する。

各seedについて:
- learned parameterized adjutant+exchange
- existing RuleBased adjutant+exchange

を同一deal / 同一その他policy条件でpaired評価する。

playing側にsamplingがある場合、比較間で乱数系列の公平性を担保する。

## Fixed policies / semantics

#452と完全一致させる。

- bidding: `frozen-raise-v1`
- playing: `ppo-separated-v1000`
- reward: 現行relative reward semantics（変更禁止）
- runtime phase order: `bidding -> adjutant -> kitty -> exchange -> playing`
- parameterized policy feature schema: #452そのまま

## Baseline

主比較:
1. existing RuleBased adjutant + existing RuleBased exchange
2. #452 learned parameterized adjutant + learned parameterized exchange

#452 final結果もreferenceとして同じtableへ載せる。

### Primary
- mean relative reward
- paired mean difference learned - RuleBased
- standard error
- 95% CI（paired bootstrapまたは妥当な方法）

### Secondary
- contract success rate
- mean contract margin
- Napoleon-side point cards
- declared target distribution
- adjutant choice distribution
- exchange bury-content distribution
- illegal action count
- fallback count
- abnormal/invariant failure count

### Paired diagnostics
- per-game reward difference distribution
- learned win/tie/loss against RuleBased on paired seeds（定義を明示）
- block別（例: 500 or 1000 gamesごと）のpaired mean difference

blockごとに改善方向が極端に不安定でないか確認する。

## Adoption criteria

正式artifact候補へ進める条件:

1. 新規独立holdoutでも paired mean difference > 0
2. 95% CI下限が0を明確に上回る、または同等に強い統計的根拠がある
3. #452の +1.213 と完全一致する必要はないが、改善方向と実用的な大きさが再現する
4. illegal/fallback/invariant failure = 0
5. contract success / marginに異常な退化がない

## Artifact候補化

条件を満たした場合、#452 best parameterized policyをrepo-managed artifactとして整理する。

例:
```text
benchmarks/non-playing-policies/parameterized-adjutant-exchange-v1/
```

または既存repo命名規約に合わせる。

最低限保存:
- feature schema/version
- adjutant 35 weights
- exchange 60 weights
- total 95 parameter vector
- parameter SHA-256
- #452 optimizer provenance
- #452 learning report reference
- 今回のindependent verification seed manifest/hash
- verification report
- bidding/playing dependency provenance/hash

人間が読めるJSON等をruntime source of truth候補にし、不要にPyTorch/ONNXへ変換しない。

このIssueではruntime wiring自体はまだ行わない。

## 失敗時

独立holdoutで改善が再現しない場合:
- formal artifact / Frozen化へ進まない
- #452 overfit / optimizer variance / seed dependenceを調査する
- parameter block別 ablation（副官のみ、埋札のみ）などを次Issue候補とする

このIssue内で再学習して結果を救済しない。

## Tests / verification

最低限:
1. #452 best parameter load parity
2. feature schema/version parity
3. same seed + same policy => deterministic/replay parity
4. new verification seeds vs #452全seed overlap 0
5. paired evaluator fairness
6. parameter serialize/load parity
7. illegal/fallback 0
8. artifact hash reproducibility（採用時）

## Git / PR

最新 `feature/issue-202-non-playing-ai` から専用branchを作成。

PR base:
`feature/issue-202-non-playing-ai`

main向けPRは禁止。

独立検証・artifact整理・tests後、Codex review最大4回（review後修正最大3回）。blocking/actionable指摘がなくなればReadyにして親branchへmergeする。

mainにはmergeしない。

## 完了時に必ず報告

- #452 best parameter/hash
- verification games数
- seed manifest/hash / #452 seedとのoverlap 0
- learned / RuleBased mean relative reward
- paired difference / SE / 95% CI
- block別再現性
- contract success / margin
- illegal/fallback/invariant
- #452結果との比較
- 正式artifact候補として採用するか
- artifact path/hash（採用時）
- 次にruntime wiring / Frozen化へ進めるか
- tests
- PR / merge commit / Codex review回数
- mainへmergeしていないこと

**Comment by eletim** [comment]
PR #455 を feature/issue-202-non-playing-ai へ merge して完了しました（merge commit bc107bb）。独立10,000 paired games: learned 1.21330、RuleBased -0.00470、paired diff +1.21800、SE 0.17891、95% CI [0.86734, 1.56866]。#452全122 manifests / 57,100 seedsとのoverlap 0、illegal/fallback/invariant failure 0、正式JSON artifact候補を保存済みです。mainにはmergeしていません。
